### ISSUE TITLE
Dada Mail Webhosting Requirements

### ISSUE THREAD

**Issue by frederickjh** [justingit/dada-mail#953]
Hi @justingit !

I know about the [Dada Mail Web Hosting page](https://dadamailproject.com/support/hosting.html), however my employer requires web hosting in the same country here in Europe where we are located due to data privacy laws.

Where we are currently hosting websites has Perl support but no way to install modules. The control panel is their own custom work from what I can tell this is not Cpanel.

Other than Dada Mail does not work on their hosting we are happy with their service. I had to move Dada Mail on to the company's backup server, which is not ideal.

I want to open a support ticket with the web hosting company to ask if they would make their hosting work with Dada Mail, but I am not sure exactly what to tell them needs to be installed to accomplish this. I hunted around the website and here on github hoping to find a document that lists out the requirements for a web hoster wanting to make their hosting work for Dada Mail out of the box but alas I did not find anything.

I know that you have the two Dada Mail Perl bundle modules `Bundle::DadaMail` and `Bundle::DadaMailXXL`  but I am not sure if this is all the Perl modules that would enable all functionality possible in Dada Mail, as I have these installed on the Backup server where Dada Mail is now and  the other day I was going through the menus and saw that a feature was disable because of a missing module.

Is there a list of all Perl Modules and other requirements that Dada Mail needs that could be given to a web hosting company to say this is what is needed to host Dada Mail?

Thanks again for your time and help!
Frederick

**Comment by justingit** [comment]
The system requirements stated are honestly the only requirements that need to be met. All the CPAN Perl modules that are needed for Dada Mail to function are bundled within the distribution itself, save for a few things like LWP and DBD::mysql (which are noted in that sys requirements doc). Dada Mail was created quite a long time ago, when shared hosting accounts were very limited in what resources were available, and that spirit still remains now. 

If there's anything else that's causing a problem with running the app, I'd like to know about it - if you could get more info on what's causing the app to not run, I would appreciate it. Should be in an error log, somewhere?

**Comment by frederickjh** [comment]
@justingit The log has no entries (`.dada_files/.logs/errors.txt`) from this year but much older entries from years gone by. I redirected the domain to another one for the install on the backup server. I just removed that forwarding, so I will need to wait til the DNS allows me to access it again so I can say what was not working. This is now several months ago and with so many other things taking priority I just now got back to deal with this. I will try and get back to you on this once the DNS allows me to access the site again.

**Comment by frederickjh** [comment]
This is the complete help topic on using Perl modules from the web hosting we are using:

> In order to use your Perl modules on a managed Cloud Server, install them in your user tree and define the complete path in your scripts.
>
> In this way, you can install and freely update your modules.
>
> Link to this FAQ: https://faq.infomaniak.com/1974

`cpan` is installed on the server. I used it to install `Bundle::DadaMailXXL`. Then afterward ran `instmodsh` to list the modules, but it only returned:

```bash
Installed modules are:
   Perl
   soplex::saferpay::ConfigurationSetup
   soplex::saferpay::MessageFactory
   soplex::saferpay::MessageObject
cmd?
```

So, @justingit with these pieces of information does it even seem possible to Run Dada Mail  on this hosting?

**Comment by justingit** [comment]
> So, @justingit with these pieces of information does it even seem possible to Run Dada Mail on this hosting?

No. But the data instmodsh is returning doesn't look right, does it? Especially if installing the Bundle::DadaMailXXL module produces no errors. These two things can't both be right. 

I don't know if I'm going to be the best help in debugging this particular hosting environment though. I can only tell you what Dada Mail will need, and if there's a problem, the error logs Dada Mail produce should point out that a Perl module can't be loaded in.

**Comment by frederickjh** [comment]
DNS has now propagated. The main page ( cgi-bin/dada/mail.cgi) loads OK, but the admin gives the yikes page.
![image](https://user-images.githubusercontent.com/864159/90495199-a033da80-e144-11ea-9f95-3d15078d9155.png)

I thought it strange that the `error.txt did not show any errors in the time frame since I move Dada Mail to the new hosting, as you said it should show error messages. I saw in `dada/DADA/Config.pm` that there is a line that tries to set the `error.txt` log file permissions to 0777. So, I checked the permissions on that file and they were not 0777. After chmod-ing the file to the correct permissions I then got an error in the log file.

> Dada Mail 9.1.1 Stable 02/14/16 Error - Cannot open list lock file `/home/clients/4f5be84d6164c3571062986d583839a9/meos.ch-mailliste/.dada_files/.tmp/_simple_auth_string_statedb.lock` - Permission denied at `DADA/Template/Widgets.pm` line 694.

Yes, I know it is an older version of Dada Mail. I have updated the version on the backup server to the newest version. Somehow no error messages, no access to the admin interface, and the time pressure to move webhosting pushed me to find a quick solution to keeping the Dada Mail install running.

In any case I looked at line 694 of `DADA/Template/Widgets.pm` and it was looking at `DISABLE_OUTSIDE_LOGINS`  and if it was equal to 1. I checked my .dada_files/.configs/.dada_config file and it was set to 1. Changing this to 0 allowed me to log in to the admin interface. 

The Dada Mail documentation  in the [Global Variables Configuration Guide](https://dadamailproject.com/support/documentation-11_11_0/global_variables.pod.html#DISABLE_OUTSIDE_LOGINS) says:
> If set to, 1, The only forms that will allow you to log into a Dada Mail list will be by a form supplied by Dada Mail itself. This means, you can't create a different form, outside the program to provide a way to login.

However, I am not using my own form and this worked on the previous host. Not sure what is going on here. I am going to move the current Dada Mail installation with the most current version to this hosting and see if it does the same thing.

**Comment by justingit** [comment]
I'm getting a little lost in the weeds on what exactly is the issue you're reporting. This isn't the general support forum, this is the issue tracker (bugs). 

And yes, I won't be able to debug a version of the app that  4.5 years old, except to say: upgrade.

**Comment by frederickjh** [comment]
Sorry I did not mean to turn this in to a support issue and the information I supplied did actually help identify why it is not running on this hosting, as you asked.  I think it may be hopeless in any case. It looks like the modules installed with cpan ended up in the users home folder in `perl5` folder.  After copying the 11.0.1 install over and fixing the config of file paths and domain names. Nothing works and I have no error reporting both from the system control panel and in Dada Mail's `error.txt`. 

> Error message: malformed header from script 'cgiwrap': Bad header: Dada Mail 11.10.1 Stable 3/24/   
> Error 500

In any case I have now been pulled away to another job, but I am still not clear on where the system requirements that you spoke of are:
> The system requirements stated are honestly the only requirements that need to be met. 
@justingit  can you give a link to them?
Thanks!
Frederick
